consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -822,19 +839,20 @@ impl PruningProofManager {
                     required_block_chain.insert(current_required_chain_block.hash);
                     selected_chain.insert(current_header.hash);
                     if required_block_chain.contains(&current_header.hash)
-                        || required_block_chain.contains(&current_required_chain_block.hash)
+                        || selected_chain.contains(&current_required_chain_block.hash)
                     {
                         intersected_with_required_block_chain = true;
                     }
                 }
 
                 if current_header.direct_parents().is_empty() // Stop at genesis
-                    || (pp_header.header.blue_score >= current_header.blue_score + required_level_0_depth
+                    // Need to ensure this does the same 2M+1 depth that block_at_depth does
```

Comment:
Is it a TODO?

Reply:
No, it's to note that the +1 here is intentional since that's how `block_at_depth` behaves. The real TODO isn't written anywhere yet which is to update `block_at_depth` to actually find the block at depth `2M` instead of block at depth `2M+1`.